school-sevice/src/test/java/liga/school/sevice/controller/SchoolControllerIntegrationTest.java

```diff
@@ -0,0 +1,306 @@
+    public void testGetStudentById() throws Exception {
+        HashSet<String> studentIds = new HashSet<>(Arrays.asList("1", "2", "3"));
+        SchoolOutComeDto schoolOutComeDto = SchoolOutComeDto.builder()
+                .id(1L)
+                .name("name12")
+                .address("address11")
+                .studentIds(studentIds)
+                .build();
+
+        Iterator<String> iterator = schoolOutComeDto.getStudentIds().iterator();
+        mockMvc.perform(get("/schools/{id}", schoolOutComeDto.getId()))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(schoolOutComeDto.getId()))
+                .andExpect(jsonPath("$.name").value(schoolOutComeDto.getName()))
+                .andExpect(jsonPath("$.address").value(schoolOutComeDto.getAddress()))
+                .andExpect(jsonPath("$.studentIds[0]").value(iterator.next()))
+                .andExpect(jsonPath("$.studentIds[1]").value(iterator.next()))
+                .andExpect(jsonPath("$.studentIds[2]").value(iterator.next()));
+    }
+
+    @Test
+    @Sql("/scripts/initSchools.sql")
+    public void testGetStudentByIdWithSchoolNotFoundException() throws Exception {
+        mockMvc.perform(get("/schools/{id}", "13"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.error").value("School not found"));
+    }
+
+    @Test
+    public void testCreateStudent() throws Exception {
+        HashSet<String> studentIds = new HashSet<>(Arrays.asList("1", "2", "3"));
+        SchoolCreateDto schoolCreateDto = SchoolCreateDto.builder()
+                .name("name12")
+                .address("address11")
+                .studentIds(studentIds)
+                .build();
+
+        when(studentFeignService.existsAllStudentsByIds(anySet())).thenReturn(true);
+        Iterator<String> iterator = schoolCreateDto.getStudentIds().iterator();
+        mockMvc.perform(put("/schools")
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON)
+                .content(mapToJson(schoolCreateDto)))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(1L))
+                .andExpect(jsonPath("$.name").value(schoolCreateDto.getName()))
+                .andExpect(jsonPath("$.address").value(schoolCreateDto.getAddress()))
+                .andExpect(jsonPath("$.studentIds[0]").value(iterator.next()))
+                .andExpect(jsonPath("$.studentIds[1]").value(iterator.next()))
+                .andExpect(jsonPath("$.studentIds[2]").value(iterator.next()));
+    }
+
+    @Test
+    public void testCreateStudentWithFalse() throws Exception {
+        HashSet<String> studentIds = new HashSet<>(Arrays.asList("1", "2", "3"));
+        SchoolOutComeDto schoolOutComeDto = SchoolOutComeDto.builder()
+                .name("name12")
+                .address("address11")
+                .studentIds(studentIds)
```

Comment:
try to remove `HashSet<String> studentIds = new HashSet<>(Arrays.asList("1", "2", "3"));`
and use static methods from classes Lists and Sets, this classes from junit or mockito and they created for using testing. And the second cause is that  `studentIds` used in one place
`.studentIds(Sets.newHashSet(Lists.newArrayList("1", "2", "3")))`

and also do it in similar places